### PR TITLE
Investigate null values in calculated properties

### PR DESCRIPTION
**Pull request by jingcheng16**
## Technical Summary

Ticket: https://dimagi.atlassian.net/browse/SAAS-15315

Salesforce sometimes would get null values from the hq, but in the next day, they can get the correct value again. I have no clue at all. So I add a`ValueError` exception to monitor when `cp_n_forms` 's value is `None`, and sent it to sentry. 
Hopefully I can get a traceback and some information regarding this.

I have deployed this branch to staging 2 weeks ago and the cp never be `None`. I will monitor sentry for production to find if we can catch anything.

### Safety story

Safe as I think `cp_n_forms` should never be `None`, it can be 0, but if it is `None`, then we raise an exception and look into it.

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/tasks.py
Comment (gherceg) [inline corehq/apps/reports/tasks.py:75]:
```suggestion
                raise ValueError(f"Null value detected for 'cp_n_forms' in domain {domain['name']}")
```

**Review by gherceg — APPROVED** [review]

**Comment by jingcheng16** [comment]
@AmitPhulera Hi Amit, we talked about this briefly. You suggested instead of raise error, I can just send it to sentry, so the task for a particular domain won't fail. But now when I relook at this, when we have None value for `cp_n_forms`, it is very likely Salesforce would receive all values None, so it doesn't matter if it fail or not. 
I plan to merge it tomorrow so I can have it deployed. Let me know what do you think!

**Comment by AmitPhulera** [comment]
> when we have None value for cp_n_forms, it is very likely Salesforce would receive all values None, so it doesn't matter if it fail or not.

Since we are using the same function for all the domains this would affect values for other domains as well. As in if there are other domains left in `domains` array which are not processes and you throw an exception the task will fail and will not generate values for remaining domains.

**Comment by jingcheng16** [comment]
@AmitPhulera The exception will be catched, and the iteration will continue, so we will still process other domains.

**Comment by jingcheng16** [comment]
@AmitPhulera Hi Amit, sorry I didn't wait for your approval, I merged it so I can have this on production for tomorrow's deploy. I also double checked with Graham, and got confirmed that other domains will still get processed.

**Comment by AmitPhulera** [comment]
Ohh! I see. You are correct. I thought that we are raising the exception at the end of the function. Thanks for getting it through 🎉

**Pull request merged**
